pkg/apis/migration/v1alpha1/migstorage_types.go

```diff
@@ -232,3 +249,62 @@ func (r *MigStorage) GetVSL(client k8sclient.Client) (*velero.VolumeSnapshotLoca
 
 	return nil, nil
 }
+
+// Get the referenced BSL Cred secret.
+func (r *MigStorage) GetBSLCredSecret(client k8sclient.Client) (*kapi.Secret, error) {
+	return GetSecret(client, r.Spec.BackupStorageConfig.CredsSecretRef)
+}
+
+// Get the referenced VSL Cred secret.
+func (r *MigStorage) GetVSLCredSecret(client k8sclient.Client) (*kapi.Secret, error) {
+	return GetSecret(client, r.Spec.VolumeSnapshotConfig.CredsSecretRef)
+}
+
+// Determine if two secrets cloud secrets are equal.
+// Returns `true` when equal.
+func (r *MigStorage) EqualsCloudSecret(a, b *kapi.Secret) bool {
+	return reflect.DeepEqual(a.Data, b.Data)
+}
+
+// Get the cloud credentials secret by labels.
+func (r *MigStorage) GetCloudSecret(client k8sclient.Client) (*kapi.Secret, error) {
+	return GetSecret(
+		client,
+		&kapi.ObjectReference{
+			Namespace: "velero",
+			Name:      "cloud-credentials",
+		})
+}
+
+// Build the cloud credentials secret.
+func (r *MigStorage) BuildCloudSecret(client k8sclient.Client) (*kapi.Secret, error) {
+	secret := &kapi.Secret{
+		ObjectMeta: metav1.ObjectMeta{
+			Labels:    CorrelationLabels(r, r.ObjectMeta.UID),
+			Namespace: "velero",
```

Comment:
How does this interact with the relevant section in `backupStorageConfig` of `mig-storage.yaml`:
```
    credsSecretRef:
      namespace: velero
      name: cloud-credentials
```
Also, where in the yaml files does the actual secret with embedded keys go? Will there be a separate creds-secret-aws.yaml added to the samples in config/samples to fill in?

Reply:
```
"How does this interact with the relevant section in backupStorageConfig of mig-storage.yaml"
```
@jortel can correct me if I'm wrong, but I don't think this directly interacts with the credsSecretRef. This is a reference in the other direction: it puts the UID of the `MigStorage` that owns this secret on the secret as a label.

````
secret:
   labels: { 'MigStorage': '<owning_migstorage_UID>' }
   [... secret content ...]

Reply:
```
"Also, where in the yaml files does the actual secret with embedded keys go? Will there be a separate creds-secret-aws.yaml added to the samples in config/samples to fill in?"
```
Yes, we'll need to add a new sample. @jortel it would be good to add a sample along with this PR, or at least shortly after.

Reply:
The credsSecretRef references a MIG secret (see: `cloud-creds.yaml`) which is used to generate the velero `cloud-credentials` secret.  I corrected the mig-storage.yaml.
@djwhatle, right.  The controller creates the velero cloud-credentials secret with the correlation label back to the mig-storage that created it.

Reply:
@djwhatle "can correct me if I'm wrong, but I don't think this directly interacts with the credsSecretRef" -- yeah, that's clearer now with the updated yaml, since it's now referencing a mig secret. With the old yaml, both secrets were velero/cloud-credentials. The update resolves my question here.